torch/csrc/api/include/torch/nn/modules/distance.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <torch/nn/cloneable.h>
+#include <torch/nn/functional/distance.h>
+#include <torch/nn/options/distance.h>
+#include <torch/nn/pimpl.h>
+#include <torch/types.h>
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+namespace torch {
+namespace nn {
+
+/// Returns the cosine similarity between :math:`x_1` and :math:`x_2`, computed
+/// along `dim`.
+struct TORCH_API CosineSimilarityImpl : Module {
```

Comment:
we should subclass `CosineSimilarityImpl` from `public torch::nn::Cloneable<CosineSimilarityImpl>`, to match the other C++ modules.